element_array_ephys/export/nwb/nwb.py

```diff
@@ -0,0 +1,629 @@
+        )
+        shank_ids = set((probe.ProbeType.Electrode & this_probe).fetch("shank"))
+        for shank_id in shank_ids:
+            electrode_group = nwbfile.create_electrode_group(
+                name=f"probe{this_probe['probe']}_shank{shank_id}",
+                description=f"probe{this_probe['probe']}_shank{shank_id}",
+                location=insert_location,
+                device=device,
+            )
+
+            electrodes_query = (
+                probe.ProbeType.Electrode & this_probe & dict(shank=shank_id)
+            ).fetch(as_dict=True)
+            for electrode in electrodes_query:
+                nwbfile.add_electrode(
+                    group=electrode_group,
+                    filtering="unknown",
+                    imp=-1.0,
+                    x=np.nan,
+                    y=np.nan,
+                    z=np.nan,
+                    rel_x=electrode["x_coord"],
+                    rel_y=electrode["y_coord"],
+                    rel_z=np.nan,
+                    shank_col=electrode["shank_col"],
+                    shank_row=electrode["shank_row"],
+                    location="unknown",
+                    id_in_probe=electrode["electrode"],
+                    shank=electrode["shank"],
+                )
+
+
+def create_units_table(
+    session_key: dict,
+    nwbfile: pynwb.NWBFile,
+    units_query,
+    paramset_record,
+    name="units",
+    desc="data on spiking units"):
+    """
+
+    ephys.CuratedClustering.Unit::unit -> units.id
+    ephys.CuratedClustering.Unit::spike_times -> units["spike_times"]
+    ephys.CuratedClustering.Unit::spike_depths -> units["spike_depths"]
+    ephys.CuratedClustering.Unit::cluster_quality_label -> units["cluster_quality_label"]
+
+    ephys.WaveformSet.PeakWaveform::peak_electrode_waveform -> units["waveform_mean"]
+
+    Parameters
+    ----------
+    session_key: dict
+    nwbfile: pynwb.NWBFile
+    units_query: ephys.CuratedClustering.Unit
+    paramset_record: int
+    name: str, optional
+        default="units"
+    desc: str, optional
+        default="data on spiking units"
+    """
+
+    # electrode id mapping
+    mapping = get_electrodes_mapping(nwbfile.electrodes)
+
+    units_table = pynwb.misc.Units(name=name, description=desc)
+    for additional_attribute in ["cluster_quality_label", "spike_depths"]:
+        units_table.add_column(
+            name=units_query.heading.attributes[additional_attribute].name,
+            description=units_query.heading.attributes[additional_attribute].comment,
+            index=additional_attribute == "spike_depths",
+        )
+
+    clustering_query = (
+        ephys.EphysRecording * ephys.ClusteringTask & session_key & paramset_record
+    )
+
+    for unit in tqdm(
+        (ephys.CuratedClustering.Unit & clustering_query).fetch(as_dict=True),
+        desc=f"creating units table for paramset {paramset_record['paramset_idx']}",
+    ):
```

Comment:
For a given `session_key` and `paramset_idx`, this loop would add all units from all rounds of curations to the same `units_tables`, so perhaps instead we should iterate through the `curation_id`?  Please see my comments in the `add_ephys_units_to_nwb` function.

Reply:
Yes, I think you are right. I'll look into making this adjustment

Reply:
My mistake.  Please ignore this suggestion.  For this export function, we are using the `ephys_no_curation` module which does not allow for manual curations and thus does not have a `curation_id`.